ntbs-service/Models/ModelBase.cs

```diff
@@ -0,0 +1,18 @@
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using ntbs_service.Models.Enums;
+
+namespace ntbs_service.Models
+{
+    public class ModelBase
+    {
+        [NotMapped]
+        public bool ShouldValidateFull { get; set; }
+
+        public void SetFullValidation(NotificationStatus notificationStatus, bool isBeingSubmitted) 

```

Comment:
I would wack a comment here explaining why this is even needed. Perhaps link out to the validation table confluence page, too?